Extend iBob docs with 4.0 features (#460)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/graphgists/query-tuning/advanced-query-tuning-example.asciidoc

Predicates that can be used to enable this optimization are:

* Existence (`WHERE exists(n.name)`)
* Equality (e.g. `WHERE n.name = 'Tom Hanks'`)
* Range (eg. `WHERE n.uid > 1000 AND n.uid < 2000`)
* Prefix (eg. `WHERE n.name STARTS WITH 'Tom'`)
* Suffix (eg. `WHERE n.name ENDS WITH 'Hanks'`)
* Substring (eg. `WHERE n.name CONTAINS 'a'`)

> * Several predicates of the above types with `OR` in between them, given that all of them are on the same property (eg. `WHERE n.prop < 10 OR n.prop = 'infinity'` )

Reviewer: I think the text 'with `OR` in between them' could be better written as 'combined using `OR`'
```suggestion
* Several predicates of the above types combined using `OR`, given that all of them are on the same property (eg. `WHERE n.prop < 10 OR n.prop = 'infinity'` )
```

> If there is an existence constraint (e.g. `CREATE CONSTRAINT constraint_name ON (p:Person) ASSERT exists(p.name)`) on the property, no predicate is required to trigger the optimization.

Reviewer: Perhaps this should be in a NOTE to highlight it?

> ==== Index-backed property-lookup for aggregating functions

Reviewer: With all these new sections, it would be nice to have a short table of contents at the top of the file, and perhaps a sub-list in the parent index page.

We also see that the `Order` column contains `p.name ASC` from the index seek operation, meaning that the rows are ordered by `p.name` in ascending order.

> Even if sorting is needed, like in the first example, the Cypher planner can utilize the knowledge about the _required order_ after each operator to move the `Sort` operator to a place with less cardinality.

Reviewer: ```suggestion
Even if sorting is needed, like in the first example, the Cypher planner can utilize knowledge of _required order_ after each operator to plan the `Sort` operator at a point in the plan with optimal cardinality.
```
